### PR TITLE
added proper support for logging in for mediawiki 1.28+

### LINKED ISSUE #42: Login does not work with Mediawiki 1.28+
It seams as since Mediawiki 1.27 there are new api's for login. Could you please add support for those new api's?

### PR DESCRIPTION
**Pull request by MananJethwani**
this resolves #42

### CODE REVIEW

**Review by kelson42 — CHANGES_REQUESTED** [review]
I believe:
* If this does not work with all user/password, then the documentation should be updated
* I see no code to secure that all requests done later to the api are using the cookie.

File: src/MediaWiki.ts [hunk 1]
```diff
@@ -54,24 +56,37 @@ class MediaWiki {
...
+        },
+        method: 'POST',
+      })
+      .then((resp) => {
+        if (resp.data.login.result !== 'Success') {
           throw new Error('Login Failed');
         }
-        downloader.loginCookie = `${jsonResponse.cookieprefix}_session=${jsonResponse.sessionid}`;
-      }
+        downloader.arrayBufferRequestOptions.headers.Cookie = resp.headers['set-cookie'].join(';');
```
Comment (MananJethwani) [inline src/MediaWiki.ts:85]: @kelson42 I set cookies here so that all request to api are being made properly was I supposed to do anything else?

Comment (kelson42) [reply src/MediaWiki.ts:85]: @MananJethwani Which piece o code secures this cookie is use for all requests?

Comment (MananJethwani) [reply src/MediaWiki.ts:85]: ` downloader.arrayBufferRequestOptions.headers.Cookie = resp.headers['set-cookie'].join(';');`
I am not sure that this works yet still checking on private wiki but mostprobably should

Comment (MananJethwani) [reply src/MediaWiki.ts:85]: sorry my bad should have changed `downloader.loginCookie`

File: src/MediaWiki.ts [hunk 2]
```diff
@@ -54,24 +56,40 @@ class MediaWiki {
...
+      })
+      .then((resp) => {
+        if (resp.data.login.result !== 'Success') {
           throw new Error('Login Failed');
         }
-        downloader.loginCookie = `${jsonResponse.cookieprefix}_session=${jsonResponse.sessionid}`;
-      }
+        downloader.arrayBufferRequestOptions.headers.cookie = resp.headers['set-cookie'].join(';');
+        downloader.jsonRequestOptions.headers.cookie = resp.headers['set-cookie'].join(';');
+        downloader.streamRequestOptions.headers.cookie = resp.headers['set-cookie'].join(';');
```
Comment (MananJethwani) [inline src/MediaWiki.ts:87]: had to add this all for securing cookies in all requests after login

File: src/mwoffliner.lib.ts [hunk 3]
```diff
@@ -187,6 +187,9 @@ async function execute(argv: any) {
     webp,
   });
 
+  /* perform login */
+  await mw.login(downloader);
```
Comment (MananJethwani) [inline src/mwoffliner.lib.ts:191]: shifted this here as in many cases site info might be restricted as well without login and login should be done before making most basic requests

Comment (kelson42) [reply src/mwoffliner.lib.ts:191]: Good

**Review by kelson42 — APPROVED** [review]

**Pull request merged**